### DISCUSSION TITLE
Addons Order? Does matter?

### DISCUSSION THREAD

**Discussion by hectorAguero** [widgetbook/widgetbook#1299] [Q&A]
I'm using these addons, but not sure which order should I follow for them, maybe some issues can arise from this, and I didn't an example with that many addons, even the ones that come with the package

```dart
        addons: [
          TextScaleAddon(min: 1.0, initialScale: 1),
          DeviceFrameAddon(
            devices: [
              Devices.ios.iPhone13Mini
            ],
            initialDevice: Devices.ios.iPhone13Mini,
          ),
          AccessibilityAddon(),
          EasyLocalizatonAddon(
            supportedLocales: [
              const Locale('en'),
              const Locale('es'),
            ],
            initialLocale: const Locale('en'),
          ),
          InspectorAddon(),
          MaterialThemeAddon(
            themes: [
              WidgetbookTheme(
                name: 'White Background',
                data: lightTheme(),
              ),
              WidgetbookTheme(
                name: 'Black Background',
                data: darkTheme(),
              ),
            ],
            initialTheme: WidgetbookTheme(
              name: 'White Background',
              data: lightTheme(),
            ),
           /// Don't work with by default, need to be switched first
          SafeAreaAddon(safeArea: false),
          AlignmentAddon(),
          ZoomAddon(),
          TimeDilationAddon(),
          ),
```

**Comment by jenshor** [comment]
Hey Hector, great question. 

Yes, the order of the Addons matters, as documented under [Addons: Order of injection and its Effect](https://docs.widgetbook.io/addons). 

To better understand the functionality of each addon it also makes sense to look into the source code which you can do by using an IDE of your choice. This will help you understand the impact to the Widget tree in better detail. 

Nonetheless, for most of the addons the order does not necessarily matter. However, for instance, the `AlignmentAddon` requires careful placement.

I'll give you an example using the `DeviceFrameAddon` and the `AlignmentAddon`. 
You can have two different orders:
1. `DeviceFrameAddon` ➡️ `AlignmentAddon`
1. `AlignmentAddon` ➡️ `DeviceFrameAddon`

The `DeviceFrameAddon` ➡️ `AlignmentAddon` approach you have is the one how it should be configured. 
If your device is for instance `800x600` in pixels you'll e.g. `Alignment.center` everything within this rectangle. 
```
.
└── DeviceFrameAddon
    └── AlignmentAddon
        └── Child
```
In general terms, this would center the child widget within the Device frame. 



The `AlignmentAddon` ➡️ `DeviceFrameAddon` approach would produce a widget tree that looks like
```
.
└── AlignmentAddon/
    └── DeviceFrameAddon/
        └── Child
```
In general terms that would mean you center a fixed-sized rectangle that contains a child. 
As the rectangle is fixed-sized there is not much to center in this case. 

Hope that helps. 
I'm happy for additional questions or follow ups 👍🏻
